state-chain/pallets/cf-environment/src/lib.rs

```diff
@@ -576,6 +619,73 @@ pub mod pallet {
 			// Witness the agg_key rotation manually in the vaults pallet for assethub
 			T::AssethubVaultKeyWitnessedHandler::on_first_key_activated(tx_id.block_number)
 		}
+
+		// TODO: PRO-2554 - User should be charged a transaction fee
+		#[pallet::call_index(10)]
+		#[pallet::weight({
+			let (dispatch_weight, dispatch_class) = weight_and_dispatch_class::<T>(calls);
+			let dispatch_weight = dispatch_weight.saturating_add(T::WeightInfo::submit_unsigned_batch_runtime_call(calls.len() as u32));
+			(dispatch_weight, dispatch_class)
+		})]
+		pub fn submit_unsigned_batch_runtime_call(
+			origin: OriginFor<T>,
+			calls: Vec<<T as Config>::RuntimeCall>,
+			transaction_metadata: TransactionMetadata,
+			user_signature_data: UserSignatureData,
+		) -> DispatchResult {
+			// unsigned extrinsic - validation happens in ValidateUnsigned
+			frame_system::ensure_none(origin)?;
+
+			// Extract signer account ID based on signature type. Validation already done in
+			// ValidateUnsigned, it should never fail to decode the signer.
+			let signer_account: T::AccountId = user_signature_data
+				.signer_account()
+				.map_err(|_| Error::<T>::FailedToDecodeSigner)?;
+
+			// Increment the account nonce to prevent replay attacks
+			frame_system::Pallet::<T>::inc_account_nonce(&signer_account);
+
+			let _ = dispatch_user_calls::<T>(
+				calls.clone(),
+				signer_account.clone(),
+				transaction_metadata.atomic,
+				T::WeightInfo::submit_unsigned_batch_runtime_call,
+			);
+
+			Ok(())
+		}
+
+		#[pallet::call_index(11)]
+		#[pallet::weight({
+			let (dispatch_weight, dispatch_class) = weight_and_dispatch_class::<T>(calls);
+			let dispatch_weight = dispatch_weight.saturating_add(T::WeightInfo::submit_batch_runtime_call(calls.len() as u32));
+			(dispatch_weight, dispatch_class)
+		})]
+		pub fn submit_batch_runtime_call(
```

Comment:
The extrinsics should be documented - basically the doc comments added to the inner functions should be here as well (or instead). This way they will appear in the metadata.